provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
This is really not the right place to open a debate about changing the
style guide. If you really want to go forward with it, I suggest to start
an email thread on prometheus-team@.

On 22 June 2016 at 10:38, Fabian Reinartz <notifications@github.com> wrote:

> In provider/mesh/peer.go
> <https://github.com/prometheus/alertmanager/pull/365#discussion_r68014440>
> :
>
> > +	"github.com/prometheus/alertmanager/types"
> > +	"github.com/prometheus/common/log"
> > +	"github.com/prometheus/common/model"
> > +	"github.com/satori/go.uuid"
> > +	"github.com/weaveworks/mesh"
> > +)
> > +
> > +// NotificationInfos provides gossiped information about which
> > +// receivers have been notified about which alerts.
> > +type NotificationInfos struct {
> > +	st     *notificationState
> > +	send   mesh.Gossip
> > +	logger log.Logger
> > +}
> > +
> > +func NewNotificationInfos(logger log.Logger) *NotificationInfos {
>
> Plenty of examples in the standard library where docs are omitted where
> they don't add any value.
>
> A few right here:
> https://golang.org/src/encoding/json/encode.go#L211
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/alertmanager/pull/365/files/656738451e34e0fcd9d8d2218133d2656718ee83#r68014440>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFWZnnu7PPp8sdc9MeST9xERgBC2z5wqks5qOPSJgaJpZM4ItRTv>
> .
>



-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales with
Company No. 6343600 | Local Branch Office | AG Charlottenburg  | HRB 110657B